### DISCUSSION TITLE
Units and formats

### DISCUSSION THREAD

**Discussion by ubitux** [harfbuzz/harfbuzz#3033] [Q&A] [answered]
Hi,

As far as I can tell, Harfbuzz has little to no documentation related to units and formats. For instance, if we look at `hb_glyph_position_t` fields (advances and offsets), it appears API users do a `/64` on them. I'm assuming this means these fields are in 26.6 *format*. Is this always true? What's the unit now? Pixels, Points, "Font units"? Does that depend on some configuration?

Similarly, I looked at `hb_glyph_extents_t`, and I typically compared the `{x,y}_bearing` fields with `FT_GlyphSlot.bitmap_{left,top}`. They are close but not the same. If I use HB extents instead of `bitmap_{left,top}/64` from FT, I end up with the text dancing a bit. Is it because there is a special scaling or unit at plays here?

Overall, I find it very hard to interpret the data I'm getting from Harfbuzz. Did I miss something?

**Comment by khaledhosny** [comment, marked as answer]
See https://github.com/harfbuzz/harfbuzz/issues/2714, improvements to the documentation are much welcome.
